File: src/my_robot_tutorials/scripts/robot_news_radio_transmitter.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

if __name__=='__main__':

	rospy.init_node('robot_news_radio_transmitter', anonymous=True) # anonymous
	# rospy.init_node('robot_news_radio_transmitter') # non-anonymous

	# create a publisher: (name of topic to publish to, type of message, queue size which acts like a buffe holds 10 messages)
	pub = rospy.Publisher("/robot_news_radio", String, queue_size=10)

	rate = rospy.Rate(2) # 2 Hz

	while not rospy.is_shutdown():
		msg = String()
		msg.data = "Hi, this is Dan from Robot News Radio!"
		pub.publish(msg)
		rate.sleep()

	rospy.loginfo("Node was stopped")
	

File: src/my_robot_tutorials/src/robot_news_radio_transmitter.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

int main(int argc, char* argv[])
{
	// init node
	ros::init(argc, argv, "robot_news_radio_transmitter", ros::init_options::AnonymousName); // anonymous
	//ros::init(argc, argv, "robot_news_radio_transmitter"); // non-anonymous
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::String>("/robot_news_radio", 10);

	ros::Rate rate(3); // Hz

	while(ros::ok())
	{
		std_msgs::String msg;
		msg.data = "Hi, this is William from the robot news radio";
		pub.publish(msg);
		rate.sleep();
	}
}

/*

Add these 2 lines to 'my_robot_tutorials' CMakeLists.txt:

	add_executable(robot_news_radio_transmitter src/robot_news_radio_transmitter.cpp)
	target_link_libraries(node_cpp ${catkin_LIBRARIES})

*/
